cmd/kube-apiserver/app/server.go

```diff
@@ -581,7 +581,15 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 		storageFactory.SetEtcdLocation(groupResource, servers)
 	}
 
-	if s.Etcd.EncryptionProviderConfigFilepath != "" {
+	if len(s.Etcd.EncryptionProviderConfigFilepath) != 0 {
+		if s.CloudProvider != nil {
```

Comment:
we have a single place we build apiserver config... hoist the cloud provider init to `CreateKubeAPIServerConfig` after Validate(), before we build any config (e.g. the call to BuildGenericConfig) (and can add doc to BuildStorageFactory that any KMS plugins must already be registered)